compiler/rustc_builtin_macros/src/format.rs

```diff
@@ -582,6 +596,68 @@ fn report_missing_placeholders(
         })
     };
 
+    let placeholders = pieces
+        .iter()
+        .filter_map(|piece| {
+            if let parse::Piece::NextArgument(argument) = piece && let ArgumentNamed(binding) = argument.position {
+                let span = fmt_span.from_inner(InnerSpan::new(argument.position_span.start, argument.position_span.end));
+                Some((span, binding))
+            } else { None }
+        })
+        .collect::<Vec<_>>();
+
+    let mut args_spans = vec![];
+    let mut fmt_spans = FxIndexSet::default();
+
+    for (i, unnamed_arg) in args.unnamed_args().iter().enumerate().rev() {
```

Comment:
We could do this with the `named_args` too? `format!("{x}", x=y)`?

Reply:
You mean a suggestion that replaces "{x}" and a named argument with "{y}"? Wouldn't that be a different diagnostic?

Reply:
`format!("{x}", x, x=y)` or `format!("{x}", x=y, x=y)` wouldn't be considered by this PR as-is, but they could be?

Reply:
The second case is covered by "duplicate argument named `x`". As for the first case, I don't know what would be the ideal error message. Currently the message is:

```
error: redundant argument
 --> src/main.rs:5:20
  |
5 |     format!("{x}", x, x=y);
  |                    ^
  |
note: the formatting specifier is referencing the binding already
 --> src/main.rs:5:15
  |
5 |     format!("{x}", x, x=y);
  |
```

if both `x` and `y` exist.